app/src/main/res/layout/activity_main.xml

```diff
@@ -4,15 +4,50 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".ui.memeitup.MemeItUpActivity">
+    android:background="@color/black"
+    tools:context=".ui.mainmemes.MainMemesActivity">
+
+    <ImageView
+        android:id="@+id/activity_main_app_title_icon"
+        style="@style/ImageAppIcon"
+        android:contentDescription="@string/icon_app"
+        app:layout_constraintBottom_toTopOf="@+id/appBarLayout"
+        app:layout_constraintEnd_toStartOf="@+id/activity_main_app_name_text_view"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
+        android:id="@+id/activity_main_app_name_text_view"
+        style="@style/TextViewAppName"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintBottom_toTopOf="@+id/appBarLayout"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@id/activity_main_app_title_icon"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <com.google.android.material.appbar.AppBarLayout
+        android:id="@+id/appBarLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="parent"
```

Comment:
if you use android:layout_width="match_parent" you dont need to set constraintEnd & constraintStart

Reply:
removed, thanks